src/services/FungibleTokens.js

```diff
@@ -36,30 +46,77 @@ export default class FungibleTokens {
         this.account = account;
     }
 
+    getParsedTokenAmount(amount, symbol, decimals) {
+        const parsedTokenAmount = symbol === 'NEAR'
+        ? parseNearAmount(amount) 
+        : parseTokenAmount(amount, decimals);
+
+        return parsedTokenAmount;
+    }
+
+    getFormattedTokenAmount(amount, symbol, decimals) {
+        const formattedTokenAmount = symbol === 'NEAR'
+        ? formatNearAmount(amount, 5) 
+        : removeTrailingZeros(formatTokenAmount(amount, decimals, 5));
+
+        return formattedTokenAmount;
+    }
+
+    async getEstimatedTotalFees(contractName, accountId) {
+        const storageBalanceAvailable = await this.isStorageBalanceAvailable(contractName, accountId);
+        if (contractName && accountId && !storageBalanceAvailable) {
+            return new BN(FT_TRANSFER_GAS).add(new BN(FT_MINIMUM_STORAGE_BALANCE)).add(new BN(FT_STORAGE_DEPOSIT_GAS)).toString();
+        } else {
+            return FT_TRANSFER_GAS;
+        }
+    }
+
+    async getEstimatedTotalNearAmount(amount) {
+        return new BN(amount).add(new BN(await this.getEstimatedTotalFees())).toString();
+    }
+
     async isStorageBalanceAvailable(contractName, accountId) {
-        return new BN((await this.getStorageBalance(contractName, accountId)).total).gte(new BN(FT_MINIMUM_STORAGE_BALANCE));
+        const storageBalance = await this.getStorageBalance(contractName, accountId);
+        return storageBalance?.total !== undefined;
     }
 
     async getStorageBalance(contractName, accountId) {
         return await this.account.viewFunction(contractName, 'storage_balance_of', { account_id: accountId });
     }
 
-    async transfer({ token: { contractName, metadata: { decimals } }, amount, receiverId, memo }) {
-        return await this.signAndSendTransaction(contractName, [
-            functionCall('ft_transfer', {
-                receiver_id: receiverId,
-                amount: parseTokenAmount(amount, decimals),
-                memo: memo
-            }, FT_TRANSFER_GAS, FT_TRANSFER_DEPOSIT)
-        ]);
+    async transfer({ contractName, parsedAmount, receiverId, memo }) {
+        if (contractName) {
+            const storageAvailable = await this.isStorageBalanceAvailable(contractName, receiverId);
+
+            if (!storageAvailable) {
+                try {
+                    await this.transferStorageDeposit(contractName, receiverId, FT_MINIMUM_STORAGE_BALANCE);
+                } catch(e) {
+                    if (e.message.includes('attached deposit is less than the mimimum storage balance')) {
```

Comment:
1. Should we re-throw errors _other_ than `attached deposit is less than the mimimum storage balance` here?
2. Type in this magic string (`mimimum` vs. `minimum`)